### ISSUE TITLE
Slow startup time on WSL2

### ISSUE THREAD

**Issue by galah92** [helix-editor/helix#7219]
### Summary

I've install Helix on WSL2 (default Ubuntu distro) through `apt`. Startup time of the editor is significantly slower then `vim`.
I imagine something is misconfigured correctly out of the box, but not sure what.
This is true for when I'm opening `hx` without a file path, and with different file types (`.rs`, `.ts`, `.txt`).

### Reproduction Steps

```
galah92@Dell14:~/adventofcode$ time hx src/main.rs
real    0m44.999s
user    0m0.807s
sys     0m0.285s
galah92@Dell14:~/adventofcode$ time vim src/main.rs

real    0m1.066s
user    0m0.054s
sys     0m0.022s
```

I think the `user` / `sys` parameters here are the relevant ones. It definitely feels like its takes 1-2 *seconds* to startup.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
galah92@Dell14:~$ cat ~/.cache/helix/helix.log
2023-06-03T13:46:13.911 helix_vcs [ERROR] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/home/galah92",
        },
    ),
}
2023-06-03T13:46:13.911 helix_vcs [ERROR] failed to open diff base for /home/galah92/.bashrc
2023-06-03T13:46:13.911 helix_vcs [ERROR] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/home/galah92",
        },
    ),
}
2023-06-03T13:46:13.911 helix_vcs [ERROR] failed to obtain current head name for /home/galah92/.bashrc
2023-06-03T13:46:14.146 helix_view::editor [ERROR] Failed to initialize the LSP for `source.bash` { cannot find binary path }
```

</details>

### Platform

Linux (WSL2)

### Terminal Emulator

Windows Terminal

### Helix Version

helix 23.03

**Comment by Tudyx** [comment]
I think it might have been fixed in https://github.com/helix-editor/helix/pull/7028
Please try to update to Helix  23.05 that contains this fix.

**Comment by galah92** [comment]
Isn't this fix specific for `.ts` files? What I'm experiencing is relevant for all file types and even when opening the editor without a specific file (empty buffer).

**Comment by Tudyx** [comment]
Yes you are right, does you still have the problem with the latest version though?

**Comment by christianfosli** [comment]
I found something similar a while ago, where it turned out to be because I had a misconfigured X-server in my WSL distro, and helix's clipboard logic was waiting for `xsel` to time out (https://github.com/helix-editor/helix/discussions/3083). Possibly relevant?

**Comment by galah92** [comment]
@christianfosli tried that just now, unfortunately it doesn't help.

**Comment by pascalkuthe** [comment]
> Isn't this fix specific for `.ts` files? What I'm experiencing is relevant for all file types and even when opening the editor without a specific file (empty buffer).

No tree sitter is the syntax highlighting used for *every* language so that PR might help altough not super likely.

**Comment by iceghost** [comment]
Apparently it is due to Helix scanning the PATH for some non-existent binary upon opening. Accessing Windows PATH inside WSL is slow, so it took quite some time for Helix to start up. Disabling Windows PATH inside WSL helped me fix this issue.

```
# inside /etc/wsl.conf
[interop]
appendWindowsPath = false
```

(But I guess, installing that missing binary could help too, but I don't know what that is)

**Comment by pascalkuthe** [comment]
We are searching for the LSP binary when opening a file. It might be reasonable to start LSPs async (since it generally involves a decent amount of I/O anyway) to speedup startupt

**Comment by the-mikedavis** [comment]
It could be the clipboard provider binary too https://github.com/helix-editor/helix/blob/d6856cfeec14305b3405a6aee90c73c80b41e48f/helix-view/src/clipboard.rs#L95-L144

**Comment by iceghost** [comment]
> It could be the clipboard provider binary too
> 
> https://github.com/helix-editor/helix/blob/d6856cfeec14305b3405a6aee90c73c80b41e48f/helix-view/src/clipboard.rs#L95-L144

I tested this by emptying `WAYLAND_DISPLAY` and `DISPLAY` to short circuit to the `win32yank.exe` branch, but it does not improve the situation.

<a href="https://asciinema.org/a/ceHCcFX04HXj1l5wLe1xhIDWD" target="_blank"><img src="https://asciinema.org/a/ceHCcFX04HXj1l5wLe1xhIDWD.svg" /></a>

(assuming that searching for more binaries takes longer than searching for one binary)

**Comment by iceghost** [comment]
> It could be the clipboard provider binary too
> 
> https://github.com/helix-editor/helix/blob/d6856cfeec14305b3405a6aee90c73c80b41e48f/helix-view/src/clipboard.rs#L95-L144

Yes, I think this is the cause! `unset DISPLAY` and `unset WAYLAND_DISPLAY` works, short-circuiting to `win32yank.exe` and makes it startup multiple time faster.

**Comment by pascalkuthe** [comment]
I guess that just means we can make searching for any binary (clipboard provider or LSP) async? It's kind of unfortunate and seems like a WSL problem gir the mist part but I guess not doing IO on the main thread is generally good practice

**Comment by matthewlloyd** [comment]
As a first-time Helix user on WSL2, this was something I noticed almost immediately. Googled and found this issue. `sudo apt install wl-clipboard` completely fixed it for me and it now starts up instantaneously, just like vim. Thank you!

**Comment by LowLevelLover** [comment]
> As a first-time Helix user on WSL2, this was something I noticed almost immediately. Googled and found this issue. `sudo apt install wl-clipboard` completely fixed it for me and it now starts up instantaneously, just like vim. Thank you!

Thanks, this solved my issue.
